pkgs/tools/misc/vector/default.nix

```diff
@@ -105,5 +108,6 @@ rustPlatform.buildRustPackage {
     homepage = "https://github.com/timberio/vector";
     license = with licenses; [ asl20 ];
     maintainers = with maintainers; [ thoughtpolice happysalada ];
+    platforms = with platforms; linux;
```

Comment:
Was there a specific reason for adding this? For `buildRustPackage`, `platforms` defaults to `platforms.linux ++ platforms.darwin` and this change makes the Darwin build "unsupported".

```
error: Package ‘vector-0.21.0’ in /nix/store/qgjhrpcas9252s88wdplvy5xayv8qqhp-source/pkgs/tools/misc/vector/default.nix:107 is not supported on ‘x86_64-darwin’, refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

        Note: For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
        (Flake) command, `--impure` must be passed in order to read this
        environment variable.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix
```